cogs/embeds.py

```diff
@@ -0,0 +1,425 @@
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+        # Provide an empty embed for every member (key)
+        self.embeds = defaultdict(lambda: EmbedData("", Embed()))
+        # Provide a default ID of -1 for every member (key) for embed fields
+        # setting it to -1 is necessary because adding embed only increments this
+        # default value of -1 ensures start on 0
+        self.embed_fields = defaultdict(lambda: -1)
+
+    @group(invoke_without_command=True, name="embed", aliases=["embedset", "set_embed"])
+    async def embed_group(self, ctx: Context) -> None:
+        """Commands for configuring the Embed messages."""
+        await ctx.send("This command is not meant to be used on its own!")
+
+    # region: basic embed settings (title, description, footer, image, color, message)
+
+    @embed_group.command(aliases=["set_title"])
+    async def title(self, ctx: Context, *, title: Unicode) -> None:
+        """Set Title for the Embed."""
+        self.embeds[ctx.author].embed.title = title
+        await ctx.send("Embeds title updated.")
+
+    @embed_group.command(aliases=["set_description"])
+    async def description(self, ctx: Context, *, description: Unicode) -> None:
+        """Set Description for the Embed."""
+        self.embeds[ctx.author].embed.description = description
+        await ctx.send("Embeds description updated.")
+
+    @embed_group.command(aliases=["add_description"])
+    async def append_description(self, ctx: Context, *, description: Unicode) -> None:
+        """Add text into Description of the Embed."""
+        self.embeds[ctx.author].embed.description += description
+        await ctx.send("Embeds description appended.")
+
+    @embed_group.command(aliases=["set_footer"])
+    async def footer(self, ctx: Context, *, footer: Unicode) -> None:
+        """Set Footer for the Embed."""
+        self.embeds[ctx.author].embed.set_footer(text=footer)
+        await ctx.send("Embeds footer updated.")
+
+    @embed_group.command(aliases=["set_image"])
+    async def image(self, ctx: Context, url: str) -> None:
+        """Set image for the Embed."""
+        self.embeds[ctx.author].embed.set_image(url=url)
+        await ctx.send("Embeds image updated.")
+
+    @embed_group.command(aliases=["set_color"])
+    async def color(self, ctx: Context, color: ColourConverter) -> None:
+        """
+        Set color for the Embed.
+
+        `color` can be HEX color code or some of the standard colors (red, blue, ...).
+        """
+        self.embeds[ctx.author].embed.colour = color
+        await ctx.send("Embeds color updated.")
+
+    @embed_group.command(aliases=["content", "msg"])
+    async def message(self, ctx: Context, *, message: str) -> None:
+        """Set message content for the Embed."""
+        self.embeds[ctx.author] = EmbedData(message, self.embeds[ctx.author].embed)
+        await ctx.send("Message content updated.")
+
+    # endregion
+    # region: author settings
+
```

Comment:
```suggestion
```

Reply:
Regions should be distinctively separated from code for better readability